dali/kernels/imgproc/resample/separable_impl.h

```diff
@@ -33,35 +34,65 @@ namespace kernels {
  * if subsequent calls do not exceed previous number of samples.
  */
 template <typename OutputElement, typename InputElement,
-          typename Interface = SeparableResamplingFilter<OutputElement, InputElement>>
+          int _spatial_ndim,
+          typename Interface = SeparableResamplingFilter<OutputElement, InputElement, _spatial_ndim>
+          >
 struct SeparableResamplingGPUImpl : Interface {
   using typename Interface::Params;
   using typename Interface::Input;
   using typename Interface::Output;
-  using SampleDesc = SeparableResamplingSetup::SampleDesc;
+  using Interface::spatial_ndim;
+  using Interface::tensor_ndim;
 
+  using ResamplingSetup = BatchResamplingSetup<spatial_ndim>;
+  using SampleDesc = typename ResamplingSetup::SampleDesc;
+  static constexpr int num_tmp_buffers = ResamplingSetup::num_tmp_buffers;
   /**
    * Generates and stores resampling setup
    */
-  BatchResamplingSetup setup;
+  ResamplingSetup setup;
 
   using IntermediateElement = float;
-  using Intermediate = OutListGPU<IntermediateElement, 3>;
+  using Intermediate = OutListGPU<IntermediateElement, tensor_ndim>;
 
   /**
    * The intermediate tensor list
    */
-  Intermediate intermediate;
+  Intermediate intermediate[num_tmp_buffers];  // NOLINT
 
   void Initialize(KernelContext &context) {
     setup.Initialize();
   }
 
+  size_t GetTmpMemSize() const {
+    // temporary buffers for more than 3D can be reused in a page-flipping fashion
+    size_t max[2] = { 0, 0 };
+    for (int i = 0; i < setup.num_tmp_buffers; i++) {
+      size_t s = setup.intermediate_sizes[i];
+      if (s > max[i&1])
+        max[i&1] = s;
+    }
+    return max[0] + max[1];
+  }
+
+  size_t GetOddTmpOffset() const {
+    // temporary buffers for more than 3D can be reused in a page-flipping fashion
+    size_t max_even = 0;
+    for (int i = 0; i < setup.num_tmp_buffers; i += 2) {
+      size_t s = setup.intermediate_sizes[i];
```

Comment:
Why not just std::max?

Reply:
I somehow dislike the unnecessary self-assignment in `x = max(x, y)`